src/Howerpoint.hs

```diff
@@ -1,41 +1,65 @@
+{-# LANGUAGE RecordWildCards     #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
 module Howerpoint where
 
-import System.IO
-import Data.List.Split
-import Data.IORef
-import Control.DeepSeq (deepseq)
+import           Control.DeepSeq              (deepseq)
+import           Control.Monad                ((>=>))
+import           Data.Char                    (isLower, toLower)
+import           Data.IORef
+import           Data.List                    (isPrefixOf)
+import           Data.List.Split
+import           Data.Maybe                   (fromMaybe, isJust, mapMaybe)
 import qualified System.Console.Terminal.Size as T
-import Data.Char (isLower, toLower)
-import Data.Maybe (isJust, fromMaybe, catMaybes)
-import Data.List (isPrefixOf)
-import Control.Applicative (liftA2)
+import           System.IO
 
 type Slide = [String]
 
+data Zipper a = Zipper [a] a [a]
```

Comment:
The zipper data structure is useful in this situation, because we have a one-dimensional pool of elements, with one element focused. Moving to adjacent slides is thus more efficient, so is querying the current one

(This could be made a functor, or even a comonad, but I didn't feel the need for these)